docs/design.md

```diff
@@ -0,0 +1,29 @@
+
+# Overview
+
+This repo describes the API implemented by controllers in the `open-cluster-management.io` project.
+
+The API enables a cluster to become a "Hub" that manages a set of other clusters.
+
+When a cluster becomes "managed", an agent is deployed that registers the managed cluster with the hub. Once registration is approved, the managed cluster can receive directives to apply specific Kubernetes manifests to run additional workloads.
+
+# API
+
+## Cluster Manager & Klusterlet
+
+API Group: `operator.open-cluster-management.io/v1`
+Kinds:
+- `ClusterManager`. Provides lifecycle management for the registration controller on the Hub.
+- `Klusterlet`. Provides lifecycle management for the registration and `ManifestWork` controllers on the managed cluster.
```

Comment:
I like this name, but it reads as the only "cute" name in the API and doesn't run congruent to `ClusterManager` or `ManagedCluster`.  `ManagedClusterConfig`?  I don't feel super strongly, more of an observation.

Reply:
I also dislike cute names, but I think Klusterlet is very meaningful because many folks familiar with Kube would be able to easily intuit what it meant.

would `Clusterlet` change things for you? That would align maybe slightly better with Kubelet and Borglet

Reply:
> I also dislike cute names, but I think Klusterlet is very meaningful because many folks familiar with Kube would be able to easily intuit what it meant.
> 
> would `Clusterlet` change things for you? That would align maybe slightly better with Kubelet and Borglet

Just want to make sure we go in with eyes open.  I don't truly mind klusterlet

Reply:
I'm okay with Clusterlet. The original name was aligned because of the "K" in Kubernetes/Kubelet.